go.mod

```diff
@@ -1,10 +1,10 @@
 module github.com/grafana/mimir
 
-go 1.22.11
+go 1.23.0
```

Comment:
@aknuds1 do you know why renovate is doing that? I thought, we've locked the version in its config ([code](https://github.com/grafana/mimir/blob/ef594298b0536b8234ec5925494d0499dc7cf88d/renovate.json5#L30-L36)).

Reply:
@narqo The version lock prevents Renovate from upgrading the Go version we use to build. Upgrading the _minimum_ version of Go to v1.23 should be fine, since v1.24 is out. The minimum and toolchain versions in go.mod doesn't affect which version we build with (that's defined in the build image).